.test-infra/jenkins/PrecommitBuilder.groovy

```diff
@@ -0,0 +1,123 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import common_job_properties
+
+/** This class defines PreCommitBuilder.build() helper for defining pre-comit jobs. */
+class PrecommitBuilder {
+  /** scope 'this' parameter from top-level script; used for binding Job DSL methods. */
+  Object scope
+
+  /** Base name for each post-commit suite job, i.e. 'Go'. */
+  String nameBase
+
+  /**  The Gradle task to execute. */
+  String gradleTask
+
+  /** Overall job timeout. */
+  int timeoutMins = 90
+
+  /** If defined, set of path expressions used to trigger the job on commit. */
+  List<String> triggerPathPatterns = []
+
+  /**
+   * Define a set of pre-commit jobs.
+   *
+   * @param additionalCustomization Job DSL closure with additional customization to apply to the job.
+   */
+  void build(Closure additionalCustomization = {}) {
+    defineCronJob additionalCustomization
+    defineCommitJob additionalCustomization
+    definePhraseJob additionalCustomization
+  }
+
+  /** Create a pre-commit job which runs on a daily schedule. */
+  private void defineCronJob(Closure additionalCustomization) {
+    def job = createBaseJob 'Cron'
+    job.with {
+      description buildDescription('on a daily schedule.')
+      common_job_properties.setAutoJob delegate
+    }
+    job.with additionalCustomization
+  }
+
+  /** Create a pre-commit job which runs on every commit to a PR. */
+  private void defineCommitJob(Closure additionalCustomization) {
+    def job = createBaseJob 'Commit', true
+    def defaultPathTriggers = [
+      '^build.gradle$',
+      '^build_rules.gradle$',
+      '^gradle.properties$',
+      '^gradlew$',
+      '^gradle.bat$',
+      '^settings.gradle$'
+    ]
+    triggerPathPatterns.addAll(defaultPathTriggers)
```

Comment:
It is better to either use braces in all places, or omit them in all places uniformly. (see line 61 vs 70)